styx-scheduler-service/src/test/java/com/spotify/styx/ScheduledExecutionUtilTest.java

```diff
@@ -0,0 +1,75 @@
+/*-
+ * -\-\-
+ * Spotify Styx Scheduler Service
+ * --
+ * Copyright (C) 2016 - 2019 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx;
+
+import static com.spotify.styx.ScheduledExecutionUtil.scheduleWithJitter;
+import static com.spotify.styx.util.ClassEnforcer.assertNotInstantiable;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.closeTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.IntStream;
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.Test;
+
+public class ScheduledExecutionUtilTest {
+
+  private static final DeterministicScheduler SCHEDULER = new DeterministicScheduler();
+
+  @Test
+  public void shouldNotBeConstructable() throws ReflectiveOperationException {
+    assertTrue(assertNotInstantiable(ScheduledExecutionUtil.class));
+  }
+
+  @Test
+  public void shouldScheduleWithJitter() {
+    // Set up a runnable that records execution timings
+    var executionTimes = new ArrayList<Integer>();
+    var time = new AtomicInteger();
+    Runnable runnable = () -> executionTimes.add(time.get());
+
+    // Schedule the runnable on an executor we can control
+    scheduleWithJitter(runnable, SCHEDULER, Duration.ofSeconds(10));
+
+    // Simulate running it for 10k seconds
+    for (int i = 0; i < 10000; i++) {
+      time.set(i);
+      SCHEDULER.tick(1, SECONDS);
+    }
+
+    // Calculate timing statistics
+    var statistics = IntStream.range(0, executionTimes.size() - 1)
+        .map(i -> executionTimes.get(i + 1) - executionTimes.get(i))
+        .summaryStatistics();
+
+    assertThat(statistics.getAverage(), is(closeTo(10, 3)));
+    assertThat(statistics.getMin(), is(both(greaterThanOrEqualTo(5)).and(lessThanOrEqualTo(10))));
```

Comment:
What about making the 0.5 and 1.5 constants for lower and upper bound package private so we can use them here when computing the expected lower and upper bounds? 

```Java
    var tickInSeconds = 10;
    scheduleWithJitter(runnable, SCHEDULER, Duration.ofSeconds(tickInSeconds));
...
    assertThat(statistics.getMin(), 
             is(both(greaterThanOrEqualTo(ScheduledExecutionUtil.LOWER_BOUND_FACTOR*tickInSeconds))
                .and(lessThanOrEqualTo(tickInSeconds))));
```

Right now I had to come back to the `ScheduledExecutionUtil` because I couldn't understand why 5 and 15 was used.

Reply:
👍